src/stream/stream.service.js

```diff
@@ -41,6 +49,37 @@ function clbStream($http, $log, clbEnv, clbError, clbResultSet) {
   }
 
   /**
+   * Builds the URL options such as the from and to date
+   * as well as the page_size
+   * @memberof module:clb-stream.clbStream
+   * @param {string} url original url
+   * @param {object} options  pageSize:15, date:'2016-07-20'
+   * @return {string} Built URL
+   */
+  function buildURLOptions(url, options) {
+    // Addition of stream options e.g. date and page_size
+    var paramToken = url.indexOf("?") === -1 ? "?" : "&";
+
+    if (options.date) {
+      var _targetDate = moment(options.date);
+      var format = 'YYYY-MM-DD';
+      url += paramToken + "from=" + _targetDate.format(format);
+      url += "&to=" + _targetDate.add(1, 'day').format(format);
+      paramToken = "&";
+    }
+
+    if (options.days) {
+      url += paramToken + "days=" + options.days;
+    }
+
+    if (options.pageSize) {
+      url += paramToken + "page_size=" + options.pageSize;
```

Comment:
if options.date == false, but options.days == true, at this point paramToken == '?', no?